connector/MongoDBAtlasODBC.query.pq

```diff
@@ -1,5 +1,233 @@
-// Use this file to write queries to test your data connector
-let
-    result = MongoDBAtlasODBC.Contents("", "")
-in
-    result
+section MongoDBAtlasODBCUnitTests;
+shared UnitTests = 
+[
+    facts = 
+    {
+        Fact("GetConnectionString - Simple URI",
+          [ DRIVER = "ADF_ODBC_DRIVER", DATABASE = "db", SERVER = "localhost:27017", AUTH_SRC = "admin" ],
+          GetConnectionString("localhost", "db",  [])
+        ),
+        Fact("GetConnectionString - All options set",
+          [ DRIVER = "ADF_ODBC_DRIVER", DATABASE = "db2", SERVER = "host.com:12345", SSL = "false",  TLS = "true", AUTH_SRC = "optionauthsrc" ],
+          GetConnectionString("mongodb+srv://host.com:12345/pathauthsrc?authSource=optionauthsrc&ssl=false&tls=true", "db2", [ ApplicationName = "{Testing AppName}"])
+        ),
+          Fact("GetConnectionString - AuthSource path only",
+          [ DRIVER = "ADF_ODBC_DRIVER", DATABASE = "test", SERVER = "localhost:27017", AUTH_SRC = "pathauthsrc" ],
+          GetConnectionString("mongodb://localhost/pathauthsrc", "test", [])
+        ),
+        Fact("GetConnectionString - AuthSource option only",
+          [ DRIVER = "ADF_ODBC_DRIVER", DATABASE = "test", SERVER = "localhost:27017", AUTH_SRC = "optionauthsrc" ],
+          GetConnectionString("mongodb://localhost/?authSource=optionauthsrc", "test", [])
+        ),
+        Fact("GetConnectionString - SSL option set",
+          [ DRIVER = "ADF_ODBC_DRIVER", DATABASE = "test", SERVER = "mongodb:27018", SSL = "true", AUTH_SRC = "admin" ],
+          GetConnectionString("mongodb:27018?ssl=true", "test",[])
+        ),
+        Fact("GetConnectionString - TLS option set",
+          [ DRIVER = "ADF_ODBC_DRIVER", DATABASE = "test", SERVER = "mongodb:27018", TLS = "false", AUTH_SRC = "admin" ],
+          GetConnectionString("mongodb:27018?tls=false", "test",[])
+        ),
+        Fact("GetConnectionString - Empty database value results in error",
+          "Database value cannot be empty",
+          try GetConnectionString("localhost","", [])  catch (e)  => e[Message]
+        ),
+        Fact("GetConnectionString - Empty host value results in error",
+          "Invalid URI: The hostname could not be parsed.",
+          try GetConnectionString("","db", [])  catch (e)  => e[Message]
+        ),
+        Fact("GetConnectionString - Username/Password in URI results in error",
+          "Invalid URI",
+          try GetConnectionString("mongodb://user:pass@localhost","db", []) catch (e)  => e[Message]
+        ), 
+        Fact("GetConnectionString - Fragment values in URI results in error",
+          "Invalid URI",
+          try GetConnectionString("mongodb://localhost/admin#fragment","db", [])  catch (e)  => e[Message]
+        )
+    },
+    report = Facts.Summarize(facts)
+][report];
+
+GetExpectedError = (function as any) as text =>
+  let
+    Error = try function
+            catch (e) => e[Message]
+  in
+    Error;
+
+
+/// COMMON UNIT TESTING CODE 

```

Comment:
This code from here to the end of the file is part of the [UnitTesting example](https://github.com/microsoft/DataConnectors/blob/master/samples/UnitTesting/UnitTesting.query.pq) and does not require review.